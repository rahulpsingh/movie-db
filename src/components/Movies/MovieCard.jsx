import React from "react";
import styled from "styled-components";
import Modal from "../Modal";
import useModal from "../Modal/useModal";

const Flex = styled.div`
  display: flex;
`;

export const Title = styled.h3`
  font-size: 16px;
  line-height: 26px;
  font-weight: normal;
`;

const TitleWrapper = styled(Flex)`
  justify-content: center;
  align-items: center;
`;

export const Wrapper = styled.div`
  border-radius: 10px;
  overflow: hidden;
  box-shadow: 0 4px 8px 0 rgb(0 0 0 / 20%);
  position: relative;
  display: flex;
  flex-direction: column;
  height: 348px;
  width: 100%;
  cursor: pointer;
`;

export const Image = styled.div`
  width: 100%;
  height: 302px;
  background-image: url(${(props) => props.src});
  background-size: cover;
  background-position: center;
`;

export const BadgeWrapper = styled(Flex)`
  height: 34px;
  width: 34px;
  border-radius: 50%;
  left: 4%;
  top: 4%;
  background: white;
  border: 1px solid black;
  text-align: center;
  position: absolute;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

const BadgeText = styled.strong`
  font-size: 12px;
  line-height: 20px;
`;

const MovieCard = ({ movieInfo }) => {
  const { toggle, visible } = useModal();

  const { title, vote_average, poster_path } = movieInfo;
  return (
    <>
      <Wrapper onClick={toggle}>
        <Image
          src={`https://image.tmdb.org/t/p/original/${poster_path}`}
          alt="movie-poster"
        >
          <BadgeWrapper>
            <BadgeText>{vote_average}</BadgeText>
          </BadgeWrapper>
        </Image>
        <TitleWrapper>
          <Title>{title}</Title>
        </TitleWrapper>
      </Wrapper>
      <Modal toggle={toggle} visible={visible} movieInfo={movieInfo} />
    </>
  );
};

export default MovieCard;
