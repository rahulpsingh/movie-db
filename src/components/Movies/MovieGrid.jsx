import React from "react";
import styled from "styled-components";
import MovieCard from "./MovieCard";

const Flex = styled.div`
  display: flex;
`;

const MovieGridWrapper = styled(Flex)`
  width: 1200px;
  margin: 0 auto;
  flex-direction: column;
`;

const MovieCardContainer = styled.ul`
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  grid-gap: 26px;
  padding: 0;
  height: fit-content;
`;

export const Header = styled.h1`
  font-family: Inter;
  font-size: 1.5rem;
  font-weight: bold;
  padding: 3vh 0vh 2vh 0vh;
`;

const MovieGrid = ({ movieList, header }) => {
  return (
    <MovieGridWrapper>
      <Header>{header}</Header>
      <MovieCardContainer>
        {movieList.map((movie, i) => (
          <MovieCard key={i} movieInfo={movie} />
        ))}
      </MovieCardContainer>
    </MovieGridWrapper>
  );
};

export default MovieGrid;
