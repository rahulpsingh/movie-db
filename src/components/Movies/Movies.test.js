import { render } from "@testing-library/react";
import MovieCard from "./MovieCard";
import MovieGrid from "./MovieGrid";

describe("MovieCard", () => {
  it("should render the MovieCard", () => {
    const { container } = render(<MovieCard toggle={() => {}} movieInfo={{}} visible />);
    expect(container).toMatchSnapshot();
  });
});

describe("MovieGrid", () => {
  it("should render the MovieGrid", () => {
    const { container } = render(<MovieGrid movieList={[]} movieInfo={{}} visible />);
    expect(container).toMatchSnapshot();
  });
});
