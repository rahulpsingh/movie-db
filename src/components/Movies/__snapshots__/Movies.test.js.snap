// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should render a modal 1`] = `
<div>
  <div
    class="sc-kEqXSa iJcsTo"
  >
    <div
      alt="movie-poster"
      class="sc-iqAclL dfzoVg"
      src="https://image.tmdb.org/t/p/original/undefined"
    >
      <div
        class="sc-fujyAs sc-crzoAE hQJiDC fLzhTR"
      >
        <strong
          class="sc-dIsUp cmnDeP"
        />
      </div>
    </div>
    <div
      class="sc-fujyAs sc-jrsJWt hQJiDC SYsXg"
    >
      <h3
        class="sc-pNWdM iVUlag"
      />
    </div>
  </div>
</div>
`;

exports[`MovieCard should render the MovieCard 1`] = `
<div>
  <div
    class="sc-kEqXSa iJcsTo"
  >
    <div
      alt="movie-poster"
      class="sc-iqAclL dfzoVg"
      src="https://image.tmdb.org/t/p/original/undefined"
    >
      <div
        class="sc-fujyAs sc-crzoAE hQJiDC fLzhTR"
      >
        <strong
          class="sc-dIsUp cmnDeP"
        />
      </div>
    </div>
    <div
      class="sc-fujyAs sc-jrsJWt hQJiDC SYsXg"
    >
      <h3
        class="sc-pNWdM iVUlag"
      />
    </div>
  </div>
</div>
`;

exports[`MovieGrid should render a modal 1`] = `
<div>
  <div
    class="sc-bqGGPW sc-ksluID kwWAjY gpckzc"
  >
    <h1
      class="sc-fnVZcZ gqCOXr"
    />
    <ul
      class="sc-hBMUJo iWmWxD"
    />
  </div>
</div>
`;

exports[`MovieGrid should render the MovieGrid 1`] = `
<div>
  <div
    class="sc-bqGGPW sc-ksluID kwWAjY gpckzc"
  >
    <h1
      class="sc-fnVZcZ gqCOXr"
    />
    <ul
      class="sc-hBMUJo iWmWxD"
    />
  </div>
</div>
`;
